### PR TITLE
gh-151: Add Github Action for lint and unit tests

### LINKED ISSUE #151: Put linting as Github Action

### PR DESCRIPTION
**Pull request by littlewhywhat**
Closes #151

### CODE REVIEW

File: frontend/src/app/components/SnackbarAlert/SnackbarAlert.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ const alertTypeToString = alertType => {
       return 'success'
     case AlertType.ERROR:
       return 'error'
+    default:
```
Comment (petrk-salsita) [inline frontend/src/app/components/SnackbarAlert/SnackbarAlert.js:16]: What's the default branch good for?

File: frontend/src/app/modules/root/root-reducer.js [hunk 2]
```diff
@@ -77,6 +77,8 @@ const createAlertForPlayerStatusUpdate = status => {
       return new UserAlert('Successfully added!', AlertType.SUCCESS)
     case StatusType.FAILURE:
       return new UserAlert(`Failed to add player :( - ${status.error}`, AlertType.ERROR)
+    default:
```
Comment (petrk-salsita) [inline frontend/src/app/modules/root/root-reducer.js:80]: What's the default branch good for?

File: frontend/src/app/modules/root/root-reducer.js [hunk 3]
```diff
@@ -88,6 +90,8 @@ const createAlertForMatchesStatusUpdate = status => {
       return new UserAlert('Succesfully created!', AlertType.SUCCESS)
     case StatusType.FAILURE:
       return new UserAlert(`Failed to create match :( - ${status.error}`, AlertType.ERROR)
+    default:
```
Comment (petrk-salsita) [inline frontend/src/app/modules/root/root-reducer.js:93]: What's the default branch good for?

File: frontend/src/app/components/SnackbarAlert/SnackbarAlert.js (hunk 1 above)
Comment (littlewhywhat) [reply frontend/src/app/components/SnackbarAlert/SnackbarAlert.js:16]: solve warnings of create-react-app scripts. It uses internally some eslint rules when you use `react-scripts start` - extending seems to be complicated - https://create-react-app.dev/docs/setting-up-your-editor/#experimental-extending-the-eslint-config

**Pull request merged**